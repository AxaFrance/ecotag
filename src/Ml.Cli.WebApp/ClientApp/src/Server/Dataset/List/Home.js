import React from 'react';
import { Link } from 'react-router-dom';
import Title from '../../../TitleBar';
import './Home.scss';
import ItemsTable from "./ItemsTable";
import EmptyArrayManager from "../../../EmptyArrayManager";

const Home = ({ items, filters, onChangePaging, onChangeFilter }) => {
    return (
        <>
            <Title title="Page datasets" subtitle="un dataset représente un essemble de données" />
            <div className="af-home container">
                <Link className="btn af-btn af-btn--quote" to="/datasets/new">
                    <span className="af-btn__text">Nouveau dataset</span>
                </Link>
                <h1 className="af-title--content">Vos datasets actifs : {items.length}</h1>
                <div className="row row--datasets-filters">
                    <div className="col">
                        <div className="af-filter-inline">
                            <span className="af-filter-inline__title">
                                <span className="glyphicon glyphicon-filter" />
                                <span className="af-filter-inline__title-text">Filtrer par</span>
                            </span>
                            <div className="af-filter-inline__field">
                                <label className="af-form__group-label" htmlFor="inputtext1">Nom du dataset</label>
                                <div className="af-form__text">
                                    <input className="af-form__input-text" id="inputtext1" name="inputtextname" onChange={onChangeFilter} type="text" />
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
                <EmptyArrayManager items={items} emptyArrayMessage="Aucun élément">
                    <ItemsTable 
                        items={items}
                        filters={filters}
                        onChangePaging={onChangePaging}
                        onChangeFilter={onChangeFilter} />
                </EmptyArrayManager>
            </div>
        </>
    )
};

export default Home;
