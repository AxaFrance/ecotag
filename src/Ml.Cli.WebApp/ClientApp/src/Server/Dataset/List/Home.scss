@import '@axa-fr/react-toolkit-form-filter-inline/dist/filter-inline.scss';

.af-btn {
  &--quote {
    margin-bottom: 1em;
  }
  &--locked {
    background-color: green;
  }
  &--pending {
    background-color: orange;
  }
  &--locked-and-work-in-progress {
    background-color: darkgoldenrod;
  }
  &--none {
    background-color: #2a87d0;
  }
}

.row--datasets-filters {
  padding-bottom: 1.5rem;
}


.af-home__access-token{
  margin-left: 2em;
}