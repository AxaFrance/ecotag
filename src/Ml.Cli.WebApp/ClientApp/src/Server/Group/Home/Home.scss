@import '@axa-fr/react-toolkit-form-filter-inline/dist/filter-inline.scss';

.af-btn {
  &--quote {
    margin-bottom: 1em;
  }
}

.row--projects-filters{
  padding-bottom: 2rem;
}