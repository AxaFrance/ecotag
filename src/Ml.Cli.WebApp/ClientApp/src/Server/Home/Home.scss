@import '@axa-fr/react-toolkit-core/src/common/scss/core.scss';

.home {
  &__links-container {
    display: flex;
    justify-content: center;
    flex-wrap: wrap;
  }
  &__link-container {
    border-radius: 20px;
    width: 200px;
    height: 200px;
    background-color: #00008f;
    display: flex;
    justify-content: center;
    align-items: center;
    font-size: 1.4em;
    text-align: center;
    transition: 0.3s;
    &--projects {
      background-color: $color-sante;
    }
    &--datasets {
      background-color: $color-epargne;
    }
    &--groups {
      background-color: $color-habitation;
    }
    &:hover {
      opacity: 0.8;
    }
  }
  &__link {
    color: #FFFFFF;
    font-weight: 400;
    font-size: 18px;
    margin-right: 20px;
    margin-bottom: 20px;
    &:hover {
      color: #FFFFFF;
    }
  }
}
