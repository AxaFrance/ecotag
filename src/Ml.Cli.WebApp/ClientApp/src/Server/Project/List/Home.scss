@import '@axa-fr/react-toolkit-form-filter-inline/dist/filter-inline.scss';

.af-btn {
  &--quote {
    margin-bottom: 1em;
  }
}

.row--projects-filters{
  padding-bottom: 2rem;
}

.projects-af-popover__wrapper > .af-popover__wrapper {
  width: 100%;
}

.af-popover__container--project-home{
  width: 100%;
}

.projects-af-popover__wrapper .af-popover__container .af-popover__arrow::before{
  background: #939393;
}

.projects-af-popover__wrapper .af-popover__container-pop {
  background: #939393;
}

.projects-af-popover__wrapper h3{
  color: white;
}