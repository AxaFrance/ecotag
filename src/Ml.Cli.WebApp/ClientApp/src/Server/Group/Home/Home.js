import React from 'react';
import PropTypes from 'prop-types';
import Title from '../../../TitleBar';
import New from './New';

import './Home.scss';
import '../../shared/Modal/modal.scss';
import EmptyArrayManager from "../../../EmptyArrayManager";
import ItemsTable from "./ItemsTable";

const Home = ({
  items,
  numberItemsTotal,
  filters,
  loaderMode,
  onChangePaging,
  onDeleteGroup,
  fields,
  isSubmitable,
  hasSubmit,
  onChangeCreateGroup,
  onSubmitCreateGroup,
  onUpdateUser,
}) => {
  return (
    <>
      <Title subtitle="Gestion des groupes d'annotateurs" title="Page Groupes"  />
      <div className="af-home container">
        <h1 className="af-title--content">Création de groupe</h1>
        <New
          disabled={isSubmitable}
          fields={fields}
          hasSubmit={hasSubmit}
          onChangeCreateGroup={onChangeCreateGroup}
          onSubmitCreateGroup={onSubmitCreateGroup}
        />
        <h1 className="af-title--content">
          {`Affichage des groupes - il y a actuellement (${numberItemsTotal}) groupes`}
        </h1>
        <EmptyArrayManager items={items} emptyArrayMessage="Aucun élément !">
          <ItemsTable
              items={items}
              filters={filters}
              loaderMode={loaderMode}
              onChangePaging={onChangePaging}
              onDeleteGroup={onDeleteGroup}
              onUpdateUser={onUpdateUser}
          />
        </EmptyArrayManager>
      </div>
    </>
  );
};

Home.defaultProps = {
  items: [],
  loaderMode: '',
};
Home.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
      name: PropTypes.string,
      users: PropTypes.array,
    })
  ),
  fields: PropTypes.object,
  loaderMode: PropTypes.string,
  isSubmitable: PropTypes.bool,
  hasSubmit: PropTypes.bool,
  onChangeCreateGroup: PropTypes.func,
  onSubmitCreateGroup: PropTypes.func,
  onChangePaging: PropTypes.func,
  onDeleteGroup: PropTypes.func,
  onUpdateUser: PropTypes.func,
};

export default Home;
