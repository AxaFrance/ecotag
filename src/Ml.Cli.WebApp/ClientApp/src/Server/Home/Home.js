import React from "react";
import Title from "../../TitleBar";
import {Link} from 'react-router-dom';
import './Home.scss';
import withAuthentication, { Administateur, Annotateur, DataScientist } from "../withAuthentication";

export const Home = ({ user: { roles = [] } }) => (
    <div className="home">
        <Title title="Accueil" goButton={false}/>
        <div className="home__links-container">
            { roles.includes(Annotateur) ? <Link className="home__link" to="/projects">
                <div className="home__link-container home__link-container--projects">Projets</div>
            </Link> : null}
            { roles.includes(DataScientist) ? <Link className="home__link" to="/datasets">
                <div className="home__link-container home__link-container--datasets">Datasets</div>
            </Link>: null}
            { roles.includes(Administateur) ? <Link className="home__link" to="/groups">
                <div className="home__link-container home__link-container--groups">Groupes</div>
            </Link> : null}
        </div>
    </div>
);

export default withAuthentication()(Home);
