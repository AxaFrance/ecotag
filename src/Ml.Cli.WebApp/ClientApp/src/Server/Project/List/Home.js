import React from 'react';
import { Link,  } from 'react-router-dom';
import Title from '../../../TitleBar';
import './Home.scss';
import '../../shared/Modal/modal.scss';
import EmptyArrayManager from "../../../EmptyArrayManager";
import ItemsTable from "./ItemsTable";

const Home = ({ items, filters, onChangePaging, onChangeFilter, onChangeSort, onDeleteProject }) => {
  const numberItemsTotal = items && items.length ? items.length : 0;
  return (
      <>
        <Title title="Page projets" subtitle="tagger un ensemble de donnée" />
        <div className="af-home container">
          <Link className="btn af-btn af-btn--quote" to="/projects/new">
            <span className="af-btn__text">Nouveau projet</span>
          </Link>
          <h1 className="af-title--content">{`Vos projets en cours (${numberItemsTotal})`}</h1>
          <div className="row row--projects-filters">
            <div className="col">
              <div className="af-filter-inline">
              <span className="af-filter-inline__title">
                <span className="glyphicon glyphicon-filter" />
                <span className="af-filter-inline__title-text">Filtrer par</span>
              </span>
                <div className="af-filter-inline__field">
                  <label className="af-form__group-label" htmlFor="inputTextFilterProjects">
                    Nom du projet
                  </label>
                  <div className="af-form__text">
                    <input
                        className="af-form__input-text"
                        id="inputTextFilterProjects"
                        name="inputTextFilterProjects"
                        onChange={event => onChangeFilter(event.target.value)}
                        type="text"
                    />
                  </div>
                </div>
              </div>
            </div>
          </div>
          <EmptyArrayManager items={items} emptyArrayMessage="Aucun élément">
            <ItemsTable
                items={items}
                filters={filters}
                onChangePaging={onChangePaging}
                onChangeSort={onChangeSort}
                onDeleteProject={onDeleteProject}
            />
          </EmptyArrayManager>
        </div>
      </>
  );
};

export default Home;
