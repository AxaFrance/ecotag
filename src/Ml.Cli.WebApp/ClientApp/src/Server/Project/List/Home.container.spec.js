
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { render, waitFor } from '@testing-library/react';
import { HomeContainer } from './Home.container';
import {BrowserRouter as Router} from "react-router-dom";

const fetch = () => Promise.resolve({ok: true, json: () => Promise.resolve([{
    "id": "0001",
    "name": "RelevÃ© d'information",
    "classification": "Publique",
    "numberTagToDo": 10,
    "createDate": "04/04/2011",
    "typeAnnotation": "NER",
    "text": "Enim ad ex voluptate culpa non cillum eu mollit nulla ex pariatur duis. Commodo officia deserunt elit sint officia consequat elit laboris tempor qui est ex. Laborum magna id deserunt ut fugiat aute nulla in Lorem pariatur. Nostrud elit consectetur exercitation exercitation incididunt consequat occaecat velit voluptate nostrud sunt. Consectetur velit eu amet minim quis sunt in.",
  }])});

describe('Home.container', () => {
  it('HomeContainer render correctly', async () => {
    const { asFragment, getByText } = render(<Router><HomeContainer fetch={fetch} /></Router>);
    const messageEl = await waitFor(() => getByText('Publique'));
    expect(messageEl).toHaveTextContent(
      'Publique'
    );
    expect(asFragment()).toMatchSnapshot();
  });
});

