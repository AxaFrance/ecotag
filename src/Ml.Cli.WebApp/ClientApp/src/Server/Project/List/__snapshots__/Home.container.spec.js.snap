// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home.container HomeContainer render correctly 1`] = `
<DocumentFragment>
  <div
    class="af-title-bar af-title-bar--backhome"
  >
    <div
      class="container-fluid container af-title-bar__wrapper"
    >
      <a
        class="btn af-btn--circle"
        href="/"
        title="Home page"
      >
        <i
          class="glyphicon glyphicon-home"
        />
      </a>
      <h1
        class="af-title-bar__title"
      >
        Page projets
      </h1>
      <small
        class="af-title-bar__subtitle"
      >
        tagger un ensemble de donnée
      </small>
    </div>
  </div>
  <div
    class="af-home container"
  >
    <a
      class="btn af-btn af-btn--quote"
      href="/projects/new"
    >
      <span
        class="af-btn__text"
      >
        Nouveau projet
      </span>
    </a>
    <h1
      class="af-title--content"
    >
      Vos projets en cours (1)
    </h1>
    <div
      class="row row--projects-filters"
    >
      <div
        class="col"
      >
        <div
          class="af-filter-inline"
        >
          <span
            class="af-filter-inline__title"
          >
            <span
              class="glyphicon glyphicon-filter"
            />
            <span
              class="af-filter-inline__title-text"
            >
              Filtrer par
            </span>
          </span>
          <div
            class="af-filter-inline__field"
          >
            <label
              class="af-form__group-label"
              for="inputTextFilterProjects"
            >
              Nom du projet
            </label>
            <div
              class="af-form__text"
            >
              <input
                class="af-form__input-text"
                id="inputTextFilterProjects"
                name="inputTextFilterProjects"
                type="text"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <table
      class="af-table"
    >
      <thead
        class="af-table-thead"
      >
        <tr
          class="af-table__tr"
        >
          <th
            class="af-table__th af-table__th--sortable"
            classmodifier="sortable"
          >
            <span
              class="af-table__th-content"
            >
              <button
                class="af-btn af-btn--table-sorting"
              >
                <span
                  class="af-btn__text"
                >
                  Nom
                </span>
                <i
                  class="glyphicon glyphicon-sorting"
                />
              </button>
            </span>
          </th>
          <th
            class="af-table__th af-table__th--sortable"
            classmodifier="sortable"
          >
            <span
              class="af-table__th-content"
            >
              <button
                class="af-btn af-btn--table-sorting"
              >
                <span
                  class="af-btn__text"
                >
                  Classification
                </span>
                <i
                  class="glyphicon glyphicon-sorting"
                />
              </button>
            </span>
          </th>
          <th
            class="af-table__th af-table__th--sortable"
            classmodifier="sortable"
          >
            <span
              class="af-table__th-content"
            >
              <button
                class="af-btn af-btn--table-sorting"
              >
                <span
                  class="af-btn__text"
                >
                  Date création
                </span>
                <i
                  class="glyphicon glyphicon-sorting-desc"
                />
              </button>
            </span>
          </th>
          <th
            class="af-table__th af-table__th--sortable"
            classmodifier="sortable"
          >
            <span
              class="af-table__th-content"
            >
              <button
                class="af-btn af-btn--table-sorting"
              >
                <span
                  class="af-btn__text"
                >
                  Type d'annotation
                </span>
                <i
                  class="glyphicon glyphicon-sorting"
                />
              </button>
            </span>
          </th>
          <th
            class="af-table__th af-table__th--sortable"
            classmodifier="sortable"
          >
            <span
              class="af-table__th-content"
            >
              <button
                class="af-btn af-btn--table-sorting"
              >
                <span
                  class="af-btn__text"
                >
                  Tags restants
                </span>
                <i
                  class="glyphicon glyphicon-sorting"
                />
              </button>
            </span>
          </th>
          <th
            class="af-table__th af-table__th--sortable"
            classmodifier="sortable"
          >
            <span
              class="af-table__th-content af-btn__text"
            >
              Action
            </span>
          </th>
        </tr>
      </thead>
      <tbody
        class="af-table__body"
      >
        <tr
          class="af-table__tr"
        >
          <td
            class="af-table__cell"
          >
            Relevé d'information
          </td>
          <td
            class="af-table__cell"
          >
            Publique
          </td>
          <td
            class="af-table__cell"
          >
            Invalid date
          </td>
          <td
            class="af-table__cell"
          >
            NER
          </td>
          <td
            class="af-table__cell"
          >
            10
          </td>
          <td
            class="af-table__cell"
          >
            <a
              class="btn af-btn--circle"
              href="#"
              id="id"
              role="button"
              tabindex="0"
              title="Editer"
            >
              <i
                class="glyphicon glyphicon-zoom-in"
              />
            </a>
            <a
              class="btn af-btn--circle"
              href="#"
              id="removeActionId"
              role="button"
              tabindex="0"
              title="Supprimer"
            >
              <i
                class="glyphicon glyphicon-remove"
              />
            </a>
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="af-paging"
    >
      <div
        class="af-paging__limit"
      >
        <div
          class="af-paging__limit"
        >
          <form
            class="af-paging__form"
          >
            <div
              class="af-form__group"
            >
              <div
                class="col col-sm-2 col-md-2 col-lg-2 col-xl-2"
              >
                <label
                  class="af-form__group-label"
                  for="home_paging"
                >
                  Afficher
                </label>
              </div>
              <div
                class="col col-sm-10 col-md-10 col-lg-10 col-xl-10"
              >
                <div
                  class="af-form__select"
                >
                  <div
                    class="af-form__select-container"
                  >
                    <select
                      class="af-form__input-select"
                      id="home_paging"
                    >
                      <option
                        value="5"
                      >
                        5
                      </option>
                      <option
                        value="10"
                      >
                        10
                      </option>
                      <option
                        value="25"
                      >
                        25
                      </option>
                      <option
                        value="50"
                      >
                        50
                      </option>
                      <option
                        value="100"
                      >
                        100
                      </option>
                    </select>
                    <span
                      class="glyphicon glyphicon-menu-down"
                    />
                  </div>
                  <span
                    class="af-form__input-cmplt"
                  >
                    éléments
                  </span>
                </div>
              </div>
            </div>
          </form>
        </div>
      </div>
      <div
        class="af-paging__pager"
      >
        <nav
          class="af-pager"
        >
          <ul
            class="af-pager__pagination"
          >
            <li
              class="af-pager__item af-pager__item--disabled"
            >
              <span
                class="af-pager__item-nolink"
              >
                « Précédent
              </span>
            </li>
            <li
              class="af-pager__item af-pager__item--active"
            >
              <a
                class="af-pager__item-link"
                href="#"
              >
                <span>
                  1
                </span>
              </a>
            </li>
            <li
              class="af-pager__item af-pager__item--disabled"
            >
              <span
                class="af-pager__item-nolink"
              >
                Suivant »
              </span>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
